core/src/main/java/com/klaviyo/coresdk/ConfigFileUtils.kt

```diff
@@ -0,0 +1,45 @@
+package com.klaviyo.coresdk
+
+import java.io.File
+import java.util.UUID
+
+internal object ConfigFileUtils {
+    private const val CONFIG_FILE_NAME = "klaviyo-sdk-config"
+
+    private const val UUID_KEY = "uuid"
+
+    fun readOrCreateUUID(): String {
+        val file = File("${KlaviyoConfig.applicationContext.filesDir}/$CONFIG_FILE_NAME")
+
+        var uuid = findConfigValue(file, UUID_KEY)
+
+        if (uuid.isNullOrEmpty()) {
+            uuid = UUID.randomUUID().toString()
```

Comment:
This'll be unique across devices yah?

Reply:
Also toString gives you right value? Did you verify what it looks like?

Reply:
This is one of Google's recommended ways to generate and store instance IDs in Android. They should be globally unique

https://developer.android.com/training/articles/user-data-ids.html#instance-ids-guids

Also this article on alternative approaches:
https://medium.com/@ssaurel/how-to-retrieve-an-unique-id-to-identify-android-devices-6f99fd5369eb

Reply:
Also if we were to use device IDs to further identify our instance IDs, then we need to include more runtime permissions so I'd rather now. 
UUIDs are considered unique across space and time according to Google's loooong RFC on the topic: https://www.ietf.org/rfc/rfc4122.txt